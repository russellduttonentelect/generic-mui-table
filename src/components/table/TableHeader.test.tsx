import { Table } from '@mui/material';
import { render } from '@testing-library/react';
import React from 'react';
import { QueryParamProvider } from 'use-query-params';
import { TableContextProvider, TableHeader } from '.';

describe('Testing Table Header component rendering', () => {
  test('Should render Table Header correctly with sensible inputs', () => {
    const headers = [
      {
        id: 'id',
        label: 'ID'
      },
      {
        id: 'lot',
        label: 'Lot'
      },
      {
        id: 'name',
        label: 'Product Name'
      }
    ];
    const header = (
      <QueryParamProvider>
        <TableContextProvider>
          <Table>
            <TableHeader
              headers={headers}
              handleSort={jest.fn()}
              order="asc"
              orderBy="id"
            />
          </Table>
        </TableContextProvider>
      </QueryParamProvider>
    );

    const { baseElement } = render(header);
    expect(baseElement).toMatchSnapshot();
  });
});
