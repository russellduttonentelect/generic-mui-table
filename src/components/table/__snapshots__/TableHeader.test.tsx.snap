// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Table Header component rendering Should render Table Header correctly with sensible inputs 1`] = `
<body>
  <div>
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            aria-sort="ascending"
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            <span
              class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-dbuacd-MuiButtonBase-root-MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              ID
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            <span
              class="MuiButtonBase-root MuiTableSortLabel-root css-dbuacd-MuiButtonBase-root-MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Lot
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            <span
              class="MuiButtonBase-root MuiTableSortLabel-root css-dbuacd-MuiButtonBase-root-MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Product Name
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                data-testid="ArrowDownwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
    </table>
  </div>
</body>
`;
